cmd/flux/create_secret_git.go

```diff
@@ -147,11 +149,17 @@ func createSecretGitCmdRun(cmd *cobra.Command, args []string) error {
 		opts.ECDSACurve = secretGitArgs.ecdsaCurve.Curve
 		opts.Password = secretGitArgs.password
 	case "http", "https":
-		if secretGitArgs.username == "" || secretGitArgs.password == "" {
-			return fmt.Errorf("for Git over HTTP/S the username and password are required")
+		if secretGitArgs.bearerToken != "" {
+			opts.BearerToken = secretGitArgs.bearerToken
+		}
+		if (secretGitArgs.username == "" || secretGitArgs.password == "") && secretGitArgs.bearerToken == "" {
+			return fmt.Errorf("for Git over HTTP/S the username and password or bearer token is required")
 		}
 		opts.Username = secretGitArgs.username
 		opts.Password = secretGitArgs.password
+		if secretGitArgs.username != "" && secretGitArgs.password != "" && secretGitArgs.bearerToken != "" {
+			return fmt.Errorf("basic auth and bearer token cannot be used together")
```

Comment:
```suggestion
			return fmt.Errorf("credentials and bearer token cannot be used together")
```

Reply:
I would maintain username and password in the other one as these are the names of the specific flags people should start to use, where as here, the flags are already in use.

Reply:
reverted to username and password.